### DISCUSSION TITLE
Global Variables?

### DISCUSSION THREAD

**Discussion by Jenssons** [bnpr/Malt#494] [Q&A]
this might be a dump question but is there a way to do a Global Variable in the sense of say i was change a variable during a animation on alot of materials is it possible to have a Global Variable where i just change it once than changing each one in each material

**Comment by pragma37** [comment]
This is not a built-in feature yet.

There are some ways to achieve the same, though:

- Using drivers.
- Using a node group. This requires the experimental [node groups branch](https://github.com/bnpr/Malt/releases/tag/node-groups-latest), though.
- If you can code you could add a custom render node to store these parameters and add them to the Scene.shader_resources, then retrieve them from a custom shader node.

**Comment by Jenssons** [comment]
so it will be something you will add in the future? also how does node groups work so if i switch over i know what to do

**Reply by pragma37** [reply]
Maybe, but it's not planned in the short/medium term.

> also how does node groups work so if i switch over i know what to do

They're a way to create custom nodes, but using nodes instead of code.

Each node tree type has its own group type. (ie. `Mesh` and `Mesh (group)` ).
You can use a group node tree from other node trees using a `Group Node`  (`Shift+A -> Other -> Group Node`).
The 'Group Node' inputs/outputs are the same as the group node tree inputs/outputs.
You can see how custom IO works here:
https://malt3d.com/Documentation/Graphs/#custom-io
